src/aws-cpp-sdk-core/include/smithy/client/AwsSmithyClient.h

```diff
@@ -70,28 +67,32 @@ namespace client
             assert(m_authSchemeResolver);
             typename ServiceAuthSchemeResolverT::ServiceAuthSchemeParameters identityParams;
 
+            identityParams.serviceName = m_serviceName;
+            identityParams.operation = ctx.m_requestName;
+            identityParams.region = m_clientConfig.region;
+
             if (ctx.m_pRequest) {
                 // refactor once auth scheme resolver will use it's own rule set
                 const auto& epParams = ctx.m_pRequest->GetEndpointContextParams();
                 for (const auto& epParam : epParams) {
                     using ParameterType = Aws::Endpoint::EndpointParameter::ParameterType;
                     if(epParam.GetStoredType() == ParameterType::STRING)
-                        identityParams.insert({epParam.GetName(), epParam.GetStrValueNoCheck()});
+                        identityParams.additionalProperties.insert({epParam.GetName(), epParam.GetStrValueNoCheck()});
                     else if (epParam.GetStoredType() == ParameterType::BOOLEAN)
-                        identityParams.insert({epParam.GetName(), epParam.GetBoolValueNoCheck()});
+                        identityParams.additionalProperties.insert({epParam.GetName(), epParam.GetBoolValueNoCheck()});
                     else
                         assert(!"Unknown endpoint parameter!");
                 }
                 const auto& serviceParams = ctx.m_pRequest->GetServiceSpecificParameters();
                 if (serviceParams) {
                     for (const auto& serviceParam : serviceParams->parameterMap) {
-                        identityParams.insert({serviceParam.first, serviceParam.second});
+                        identityParams.additionalProperties.insert({serviceParam.first, serviceParam.second});
                     }
                 }
             }
             Aws::Vector<AuthSchemeOption> authSchemeOptions = m_authSchemeResolver->resolveAuthScheme(identityParams);
 
-            auto authSchemeOptionIt = std::find_first_of(authSchemeOptions.begin(), authSchemeOptions.end(),
+            auto authSchemeOptionIt = std::find_if(authSchemeOptions.begin(), authSchemeOptions.end(),
```

Comment:
why are we asserting "assert(authSchemeOptionIt != authSchemeOptions.end());" when the else case of auth scheme not found we return AWS error. In such a case we wont ever return error then. 
Assertions are not good and should be handled gracefully with proper error